@navikt/core/icons/package.json

```diff
@@ -38,7 +38,8 @@
     "uuid": "^8.3.2"
   },
   "peerDependencies": {
-    "react": "^16.8.0 || ^17.0.0"
+    "react": "^16.8.0 || ^17.0.0",
+    "@types/react": "^17.0.30"
```

Comment:
Funker dette om bruker har react 16?

Reply:
Litt usikker på hva vi gjør i denne casen. For ja vil nok kunne oppstå noen typefeil, men samtidig så er alle typene til komponentene bygd med @types/react v17. Noe som vil si at hvis man bruker eks @types/react v16.8 så vil det garantere feil ved bruk av loader og ikonpakke (som jeg vet om nå). 

Lurer derfor på om vi ønsker å bare sette react "^17.0.0", og da fjerne støtte for v16.8? Gir mening med tanke på at vi sikrer at komponentene er kompatible med react v18, v19 etc da (React garanterer at v17 skal være kompatibel med alle nye versjoner)

Reply:
Hmm, det er interessant. Det blir jo an avveiing mellom kostnaden ved å støtte/ikke støtte 16.8. Her kjenner du detaljene bedre enn meg. Hva skal f.eks. til for at ikonpakken også er 16.8-kompatibel? Er det i det hele tatt realistisk?

Reply:
React forwardref ble publisert i React v16.3 (https://reactjs.org/blog/2018/03/29/react-v-16-3.html#forwardref-api) så for selve react-versjonen av ikonpakken er det ingen problem. For resten av biblioteket så er `v16.8` lavest støttet versjon på grunn av hooks.

 Problemet oppstår nå ved bruk av komponentene i typescript-prosjekt, da mer spesifikt komponenter som bruker `React.SVGProps<SVGSVGElement>`
Typedefinisjonene som blir lagt i esm-mappen `d.ts` får da denne typen
```
React.ForwardRefExoticComponent<Pick<React.SVGProps<SVGSVGElement>, ... | "fr" | ... > & React.RefAttributes<SVGSVGElement>>
```
Der "fr" ikke fantes før `@types/react: "17.0.29"`, så alle tidligere versjoner vil gi feil ved bruk. 
Vil tro det har vært lite endringer fra @types/react v16.8 -> v17 ellers så kan hende det fungerer helt fint, men nesten umulig å sjekke da det er så mange nisje typer i react

Reply:
Om mange må oppgradere fra 16.8 til 17 (det har kanskje du statistikk på?) som en konsekvens av denne endringen, så tenker jeg det kan være lurt å lufte dette på Slack. Eller er dette kanskje en mindre problem enn jeg tror?

Reply:
Gikk gjennom nesten alle repoene som nå tar i bruk ds-react/ds-css og var bare 1 av 35+ som ikke brukte react `v17`. Lager en liten undersøkelse om det neste uke på `#nav-frontend` for å dobbelsjekke, men tenker vi kan fjerne `v16.9`-støtte